R/get_pvol_se.R

```diff
@@ -0,0 +1,43 @@
+get_pvol_se_radar_mapping <- c(
+  "seang" = "angelholm", "seatv" = "atvidaberg", "sebaa" = "balsta",
+  "sehem" = "hemse", "sehuv" = "hudiksvall", "sekaa" = "karlskrona",
+  "sekrn" = "kiruna", "selek" = "leksand", "sella" = "lulea",
+  "seoer" = "ornskoldsvik", "seosd" = "ostersund", "sevax" = "vara"
+)
+
+get_pvol_se <- function(radar, time, ..., call = rlang::caller_env()) {
+  if (!radar %in% names(get_pvol_se_radar_mapping)) {
+    cli::cli_abort(
+      c(
+        x = "The radar {.val {radar}} is not found in the mapping for Swedish radars",
+        i = "Most likely this means an invalid odim code has been provided"
+      ),
+      call = call, class = "getRad_error_get_pvol_se_radar_not_found"
+    )
+  }
+  radar_name <- get_pvol_se_radar_mapping[radar]
+  url <- glue::glue('/area/{radar_name}/product/qcvol/{lubridate::year(time)}/{lubridate::month(time)}/{{lubridate::day(time)}}/radar_{radar_name}_qcvol_{strftime(time, "%Y%m%d%H%M", tz="UTC" )}.h5')
+  req <- withCallingHandlers(
+    httr2::request(
+      "https://opendata-download-radar.smhi.se/api/version/latest"
+    ) |>
+      req_user_agent_getrad() |>
+      httr2::req_url_path_append(
+        glue::glue(url)
+      ) |>
+      httr2::req_perform(path = tempfile(fileext = ".h5"), error_call = call),
+    httr2_http_404 = function(cnd) {
+      cli::cli_abort(
+        c(
+          x = "No polar volume data could be found for {.val {radar}} at time {.val {time}}",
+          i = "Volume data in Sweden is only available for 24 hours",
+          i = "If the requested time is within the last 24 hours the error might relate to a server outage or package problem"
+        ),
+        parent = cnd, call = call, class = "getRad_error_get_pvol_se_data_not_found"
+      )
+    }
+  )
+  pvol <- bioRad::read_pvolfile(req$body, ...)
+  file.remove(req$body)
```

Comment:
If the function fails before this line, the file will not be removed. Do we need to write to drive or can we do this in memory? 

See also: https://withr.r-lib.org/reference/with_file.html

Reply:
As `httr2` does the file writing and `bioRad` reading I don't think it will work with connections, neither does `rhdf5`. The with file is a nice suggestion, it would maybe require an extra dependency? Lets do that in a separate issue as it is something that would need to change throughout.

Reply:
see #98

Reply:
Let's make an issue about it. Especially since we are writing a large file we should do everything to avoid leaving stuff in temp dirs even by accident. 

I recently read this: https://contributor.r-project.org/cran-cookbook/code_issues.html#problem-5